### PR TITLE
discovery: log error only if both ssl and non-ssl srv lookups fail

### LINKED ISSUE #3414: Incorrect dns discovery log - discovery: error querying DNS SRV records for _etcd-server lookup _etcd-server._tcp.etcd.dev.com
I have a 3 node cluster and 1 proxy running on node01.

Logs in all the nodes and proxy shows below line as error
**discovery: discovery: error querying DNS SRV records for _etcd-server lookup _etcd-server._tcp.etcd.dev.com on 10.12.12.76:53: no such host**

Why should this be an error?
I wanted to bootstrap on ssl
[Documentaion](https://github.com/coreos/etcd/blob/master/Documentation/clustering.md#dns-discovery) says I can either use **_etcd-server._tcp.example.com.** or **_etcd-server-ssl._tcp.example.com** and if ssl entry is found then etcd will attempt the bootstrapping process over SSL.

Agree, that bootstrapping using DNS SRV should be either on ssl or non-ssl.

But it complains either way, If I bootstrap over non-ssl then it complains the other way (discovery: discovery: error querying DNS SRV records for _etcd-server lookup **_etcd-server-ssl._tcp.etcd.dev.com** on 10.12.12.76:53: no such host). I think this should not be an error as I dont think a cluster should be bootstrapped on both ssl and non-ssl. It should be either ssl or non-ssl as the documentation says.

### PR DESCRIPTION
**Pull request by raoofm**
discovery: log error only if both ssl and non-ssl srv lookups fail
Earlier we were logging as soon as one of the lookups failed.

Fixes #3414

### CODE REVIEW

File: discovery/srv.go [hunk 1]
```diff
@@ -77,16 +77,19 @@ func SRVGetCluster(name, dns string, defaultToken string, apurls types.URLs) (st
 
 	failCount := 0
 	err := updateNodeMap("etcd-server-ssl", "https://")
+	srvErr := make([]string, 2)
 	if err != nil {
-		plog.Warningf("error querying DNS SRV records for _etcd-server-ssl %s", err)
+		srvErr[0] = fmt.Sprintf("error querying DNS SRV records for _etcd-server-ssl %s", err);
```
Comment (xiang90) [inline discovery/srv.go]: remove the trailing `;`.

File: discovery/srv.go [hunk 2]
```diff
@@ -77,16 +77,19 @@ func SRVGetCluster(name, dns string, defaultToken string, apurls types.URLs) (st
...
 		failCount += 1
 	}
 	err = updateNodeMap("etcd-server", "http://")
 	if err != nil {
-		plog.Warningf("discovery: error querying DNS SRV records for _etcd-server %s", err)
+		srvErr[1] = fmt.Sprintf("error querying DNS SRV records for _etcd-server %s", err);
```
Comment (xiang90) [inline discovery/srv.go]: remove the trailing `;`

File: discovery/srv.go [hunk 3]
```diff
@@ -77,16 +77,19 @@ func SRVGetCluster(name, dns string, defaultToken string, apurls types.URLs) (st
...
 		failCount += 1
 	}
 	if failCount == 2 {
+		plog.Warningf(srvErr[0]);
```
Comment (xiang90) [inline discovery/srv.go]: remove the trailing ;

File: discovery/srv.go [hunk 4]
```diff
@@ -77,16 +77,19 @@ func SRVGetCluster(name, dns string, defaultToken string, apurls types.URLs) (st
...
+		plog.Warningf(srvErr[1]);
```
Comment (xiang90) [inline discovery/srv.go]: remove the trailing ;

**Comment by xiang90** [comment]
The change looks good to me. But can you please squash your commits into one? (I would suggest you: 1. pull original master 2. checkout a new branch 3. apply the changes 4. push to github and send the pull request.)

**Comment by raoofm** [comment]
Closing this to create a new one as per suggestion.

**Pull request closed without merging**